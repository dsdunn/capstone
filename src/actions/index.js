export const signIn = (user) => ({
  type: 'SIGN_IN',
  user
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const updateUser = (user) => ({
  type: 'UPDATE_USER',
  user
})

export const setProfile = profile => ({
  type: 'SET_PROFILE',
  profile
})

export const addCollections = (collections) => ({
  type: 'ADD_COLLECTIONS',
  collections
})

export const setCollection = collection => ({
  type: 'SET_COLLECTION',
  collection
})

export const updateCollectionsList = (collectionsList) => {
  type: 'UPDATE_COLLECTIONS_LIST',
  collectionsList
}