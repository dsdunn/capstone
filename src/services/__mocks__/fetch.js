export const getAllCollections = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'string',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getUserCollections = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'string',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getCollectionsByCategory = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getCollection = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getUserInfo = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666
    }])
  })

export const getSearchResults = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      collections:[]
    }])
  })

export const postCollection = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'
    }])
  })