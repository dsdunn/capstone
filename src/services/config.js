export default {
  apiKey: '66642069',
  url: 'https://collecshare.herokuapp.com/api/v1'
}