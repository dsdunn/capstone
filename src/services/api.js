import config from './config';

export const apiFetch = (path, method = 'GET', payload = null) => {
  let body = payload ? JSON.stringify(payload) : null
  return fetch(`${config.url}${path}`, {
    method,
    headers: {
     'Content-Type': 'application/json',
     'Accept': 'application/json'
    },
    body
  })
}

export const apiFormDataFetch = (path, method = 'PUT', payload = null) => {
  console.log(path, method, payload);
  let body = payload; 
  return fetch(`${config.url}${path}`, {
    method,
    body
  })
}
