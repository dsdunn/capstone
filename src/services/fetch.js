import { apiFetch, apiFormDataFetch } from './api'

export const postUserInfo = user => {
  return apiFetch('/users', 'POST', user)
}

export const getUserInfo = uid => {
  return apiFetch(`/users/${uid}`, 'GET')
  .then(response => response.json())
}

export const putUserInfo = (userInfo, uid) => {
  return apiFormDataFetch(`/users/${uid}`, 'PUT', userInfo)
  .then(response => console.log(response))
  .then(response => response.json())
}

export const postCollection = collection => {
  return apiFetch(`/collections`, `POST`, collection)
  .then(response => response.json())
}

export const getAllCollections = () => {
  return apiFetch('/collections')
  .then(response => response.json())
}

export const getUserCollections = uid => {
  return apiFetch(`/collections?uid=${uid}`)
  .then(response => response.json())
}