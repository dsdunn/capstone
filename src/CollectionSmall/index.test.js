import React from 'react';
import { shallow } from 'enzyme';
import { CollectionSmall, mapDispatchToProps } from './'
import { setProfile, setCollection } from '../actions';
import { getUserInfo, getCollection } from '../services/fetch'

jest.mock('../services/fetch');

describe('CollectionSmall', () => {
  let wrapper;
  let mockCollection;
  let history;
  let mockSetCollection;

  beforeEach(() => {
    mockSetCollection = jest.fn()
    history = { location: 'location',
                goBack: jest.fn(),
                push: jest.fn() 
              }

    mockCollection = {
                        id: 'int',
                        uid: 'string',
                        category: 'string',
                        title: 'string',
                        description: 'string',
                        image: 'string'
                      }
    wrapper = shallow(<CollectionSmall collection={mockCollection}
                                       history={history}
                                       setCollection={mockSetCollection}/>)
  })

  it('should get a users info when mounted', () => {
    expect(getUserInfo).toHaveBeenCalled()
  })

  it('should start with default state of an empty user', () => {
    const actual = wrapper.state()
    const expected = { user: [ { id: 666 } ] }

    expect(actual).toEqual(expected)
  })

  it('should do the thing when viewCollectoinBig is called', () => {
    wrapper.instance().viewCollectionBig()
    expect(history.push).toHaveBeenCalled()
  })

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should return a props object with setProfile', () => {
    const mockDispatch = jest.fn();
    const mockUser = { password1: '123456',
                       password2: '123456',
                       email: 'heck@yeah.com',
                       username: 'Sparkle' }

    const mappedProps = mapDispatchToProps(mockDispatch)
    const actionToDispatch = setProfile(mockUser);

    mappedProps.setProfile(mockUser);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

})