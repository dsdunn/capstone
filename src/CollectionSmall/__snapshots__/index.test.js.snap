// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionSmall should match the snapShot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CollectionSmall
    collection={
        Object {
            "category": "string",
            "description": "string",
            "id": "int",
            "image": "string",
            "title": "string",
            "uid": "string",
          }
    }
    history={
        Object {
            "goBack": [Function],
            "location": "location",
            "push": [Function],
          }
    }
    setCollection={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="collection-small"
>
        <img
                className="collection-small-image"
                src="https://collecshare.herokuapp.com/string"
        />
        <div
                className="collection-small-top"
        >
                <div
                        className="collection-small-header"
                >
                        <div>
                                <h3
                                        className="collection-small-title"
                                >
                                        string
                                </h3>
                                <div
                                        className="collection-small-category"
                                >
                                        CATEGORY: 
                                        <span
                                                className="collection-small-category-span"
                                        >
                                                string
                                        </span>
                                </div>
                        </div>
                        <div
                                className="user-link collection-small-user-display "
                                onClick={[Function]}
                        >
                                <div>
                                        <p
                                                className="user-link collection-small-username"
                                        />
                                        <p
                                                className="user-link collection-small-location"
                                        >
                                                earth
                                        </p>
                                </div>
                                <span>
                                        <img
                                                className="user-link collection-small-avatar"
                                                src="https://collecshare.herokuapp.com/undefined"
                                        />
                                </span>
                        </div>
                </div>
                <p
                        className="collection-small-description"
                >
                        string
                </p>
        </div>
</div>,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="collection-small-image"
            src="https://collecshare.herokuapp.com/string"
/>,
          <div
            className="collection-small-top"
>
            <div
                        className="collection-small-header"
            >
                        <div>
                                    <h3
                                                className="collection-small-title"
                                    >
                                                string
                                    </h3>
                                    <div
                                                className="collection-small-category"
                                    >
                                                CATEGORY: 
                                                <span
                                                            className="collection-small-category-span"
                                                >
                                                            string
                                                </span>
                                    </div>
                        </div>
                        <div
                                    className="user-link collection-small-user-display "
                                    onClick={[Function]}
                        >
                                    <div>
                                                <p
                                                            className="user-link collection-small-username"
                                                />
                                                <p
                                                            className="user-link collection-small-location"
                                                >
                                                            earth
                                                </p>
                                    </div>
                                    <span>
                                                <img
                                                            className="user-link collection-small-avatar"
                                                            src="https://collecshare.herokuapp.com/undefined"
                                                />
                                    </span>
                        </div>
            </div>
            <p
                        className="collection-small-description"
            >
                        string
            </p>
</div>,
        ],
        "className": "collection-small",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "collection-small-image",
            "src": "https://collecshare.herokuapp.com/string",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="collection-small-header"
>
                <div>
                                <h3
                                                className="collection-small-title"
                                >
                                                string
                                </h3>
                                <div
                                                className="collection-small-category"
                                >
                                                CATEGORY: 
                                                <span
                                                                className="collection-small-category-span"
                                                >
                                                                string
                                                </span>
                                </div>
                </div>
                <div
                                className="user-link collection-small-user-display "
                                onClick={[Function]}
                >
                                <div>
                                                <p
                                                                className="user-link collection-small-username"
                                                />
                                                <p
                                                                className="user-link collection-small-location"
                                                >
                                                                earth
                                                </p>
                                </div>
                                <span>
                                                <img
                                                                className="user-link collection-small-avatar"
                                                                src="https://collecshare.herokuapp.com/undefined"
                                                />
                                </span>
                </div>
</div>,
              <p
                className="collection-small-description"
>
                string
</p>,
            ],
            "className": "collection-small-top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <h3
                                        className="collection-small-title"
                    >
                                        string
                    </h3>
                    <div
                                        className="collection-small-category"
                    >
                                        CATEGORY: 
                                        <span
                                                            className="collection-small-category-span"
                                        >
                                                            string
                                        </span>
                    </div>
</div>,
                  <div
                    className="user-link collection-small-user-display "
                    onClick={[Function]}
>
                    <div>
                                        <p
                                                            className="user-link collection-small-username"
                                        />
                                        <p
                                                            className="user-link collection-small-location"
                                        >
                                                            earth
                                        </p>
                    </div>
                    <span>
                                        <img
                                                            className="user-link collection-small-avatar"
                                                            src="https://collecshare.herokuapp.com/undefined"
                                        />
                    </span>
</div>,
                ],
                "className": "collection-small-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h3
                        className="collection-small-title"
>
                        string
</h3>,
                      <div
                        className="collection-small-category"
>
                        CATEGORY: 
                        <span
                                                className="collection-small-category-span"
                        >
                                                string
                        </span>
</div>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "string",
                        "className": "collection-small-title",
                      },
                      "ref": null,
                      "rendered": "string",
                      "type": "h3",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "CATEGORY: ",
                          <span
                            className="collection-small-category-span"
>
                            string
</span>,
                        ],
                        "className": "collection-small-category",
                      },
                      "ref": null,
                      "rendered": Array [
                        "CATEGORY: ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "string",
                            "className": "collection-small-category-span",
                          },
                          "ref": null,
                          "rendered": "string",
                          "type": "span",
                        },
                      ],
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div>
                        <p
                                                className="user-link collection-small-username"
                        />
                        <p
                                                className="user-link collection-small-location"
                        >
                                                earth
                        </p>
</div>,
                      <span>
                        <img
                                                className="user-link collection-small-avatar"
                                                src="https://collecshare.herokuapp.com/undefined"
                        />
</span>,
                    ],
                    "className": "user-link collection-small-user-display ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <p
                            className="user-link collection-small-username"
/>,
                          <p
                            className="user-link collection-small-location"
>
                            earth
</p>,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": undefined,
                            "className": "user-link collection-small-username",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "p",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "earth",
                            "className": "user-link collection-small-location",
                          },
                          "ref": null,
                          "rendered": "earth",
                          "type": "p",
                        },
                      ],
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <img
                          className="user-link collection-small-avatar"
                          src="https://collecshare.herokuapp.com/undefined"
/>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "user-link collection-small-avatar",
                          "src": "https://collecshare.herokuapp.com/undefined",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "img",
                      },
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "string",
                "className": "collection-small-description",
              },
              "ref": null,
              "rendered": "string",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="collection-small"
>
          <img
                    className="collection-small-image"
                    src="https://collecshare.herokuapp.com/string"
          />
          <div
                    className="collection-small-top"
          >
                    <div
                              className="collection-small-header"
                    >
                              <div>
                                        <h3
                                                  className="collection-small-title"
                                        >
                                                  string
                                        </h3>
                                        <div
                                                  className="collection-small-category"
                                        >
                                                  CATEGORY: 
                                                  <span
                                                            className="collection-small-category-span"
                                                  >
                                                            string
                                                  </span>
                                        </div>
                              </div>
                              <div
                                        className="user-link collection-small-user-display "
                                        onClick={[Function]}
                              >
                                        <div>
                                                  <p
                                                            className="user-link collection-small-username"
                                                  />
                                                  <p
                                                            className="user-link collection-small-location"
                                                  >
                                                            earth
                                                  </p>
                                        </div>
                                        <span>
                                                  <img
                                                            className="user-link collection-small-avatar"
                                                            src="https://collecshare.herokuapp.com/undefined"
                                                  />
                                        </span>
                              </div>
                    </div>
                    <p
                              className="collection-small-description"
                    >
                              string
                    </p>
          </div>
</div>,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="collection-small-image"
              src="https://collecshare.herokuapp.com/string"
/>,
            <div
              className="collection-small-top"
>
              <div
                            className="collection-small-header"
              >
                            <div>
                                          <h3
                                                        className="collection-small-title"
                                          >
                                                        string
                                          </h3>
                                          <div
                                                        className="collection-small-category"
                                          >
                                                        CATEGORY: 
                                                        <span
                                                                      className="collection-small-category-span"
                                                        >
                                                                      string
                                                        </span>
                                          </div>
                            </div>
                            <div
                                          className="user-link collection-small-user-display "
                                          onClick={[Function]}
                            >
                                          <div>
                                                        <p
                                                                      className="user-link collection-small-username"
                                                        />
                                                        <p
                                                                      className="user-link collection-small-location"
                                                        >
                                                                      earth
                                                        </p>
                                          </div>
                                          <span>
                                                        <img
                                                                      className="user-link collection-small-avatar"
                                                                      src="https://collecshare.herokuapp.com/undefined"
                                                        />
                                          </span>
                            </div>
              </div>
              <p
                            className="collection-small-description"
              >
                            string
              </p>
</div>,
          ],
          "className": "collection-small",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "collection-small-image",
              "src": "https://collecshare.herokuapp.com/string",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="collection-small-header"
>
                  <div>
                                    <h3
                                                      className="collection-small-title"
                                    >
                                                      string
                                    </h3>
                                    <div
                                                      className="collection-small-category"
                                    >
                                                      CATEGORY: 
                                                      <span
                                                                        className="collection-small-category-span"
                                                      >
                                                                        string
                                                      </span>
                                    </div>
                  </div>
                  <div
                                    className="user-link collection-small-user-display "
                                    onClick={[Function]}
                  >
                                    <div>
                                                      <p
                                                                        className="user-link collection-small-username"
                                                      />
                                                      <p
                                                                        className="user-link collection-small-location"
                                                      >
                                                                        earth
                                                      </p>
                                    </div>
                                    <span>
                                                      <img
                                                                        className="user-link collection-small-avatar"
                                                                        src="https://collecshare.herokuapp.com/undefined"
                                                      />
                                    </span>
                  </div>
</div>,
                <p
                  className="collection-small-description"
>
                  string
</p>,
              ],
              "className": "collection-small-top",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      <h3
                                            className="collection-small-title"
                      >
                                            string
                      </h3>
                      <div
                                            className="collection-small-category"
                      >
                                            CATEGORY: 
                                            <span
                                                                  className="collection-small-category-span"
                                            >
                                                                  string
                                            </span>
                      </div>
</div>,
                    <div
                      className="user-link collection-small-user-display "
                      onClick={[Function]}
>
                      <div>
                                            <p
                                                                  className="user-link collection-small-username"
                                            />
                                            <p
                                                                  className="user-link collection-small-location"
                                            >
                                                                  earth
                                            </p>
                      </div>
                      <span>
                                            <img
                                                                  className="user-link collection-small-avatar"
                                                                  src="https://collecshare.herokuapp.com/undefined"
                                            />
                      </span>
</div>,
                  ],
                  "className": "collection-small-header",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <h3
                          className="collection-small-title"
>
                          string
</h3>,
                        <div
                          className="collection-small-category"
>
                          CATEGORY: 
                          <span
                                                    className="collection-small-category-span"
                          >
                                                    string
                          </span>
</div>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "string",
                          "className": "collection-small-title",
                        },
                        "ref": null,
                        "rendered": "string",
                        "type": "h3",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            "CATEGORY: ",
                            <span
                              className="collection-small-category-span"
>
                              string
</span>,
                          ],
                          "className": "collection-small-category",
                        },
                        "ref": null,
                        "rendered": Array [
                          "CATEGORY: ",
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "string",
                              "className": "collection-small-category-span",
                            },
                            "ref": null,
                            "rendered": "string",
                            "type": "span",
                          },
                        ],
                        "type": "div",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <div>
                          <p
                                                    className="user-link collection-small-username"
                          />
                          <p
                                                    className="user-link collection-small-location"
                          >
                                                    earth
                          </p>
</div>,
                        <span>
                          <img
                                                    className="user-link collection-small-avatar"
                                                    src="https://collecshare.herokuapp.com/undefined"
                          />
</span>,
                      ],
                      "className": "user-link collection-small-user-display ",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <p
                              className="user-link collection-small-username"
/>,
                            <p
                              className="user-link collection-small-location"
>
                              earth
</p>,
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": undefined,
                              "className": "user-link collection-small-username",
                            },
                            "ref": null,
                            "rendered": null,
                            "type": "p",
                          },
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "earth",
                              "className": "user-link collection-small-location",
                            },
                            "ref": null,
                            "rendered": "earth",
                            "type": "p",
                          },
                        ],
                        "type": "div",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <img
                            className="user-link collection-small-avatar"
                            src="https://collecshare.herokuapp.com/undefined"
/>,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "className": "user-link collection-small-avatar",
                            "src": "https://collecshare.herokuapp.com/undefined",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "img",
                        },
                        "type": "span",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "string",
                  "className": "collection-small-description",
                },
                "ref": null,
                "rendered": "string",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
